homie-device/src/types.rs

```diff
@@ -59,32 +101,198 @@ impl Property {
+    }
+
+    /// Create a new boolean property with the given attributes.
+    ///
+    /// # Arguments
+    /// * `id`: The subtopic ID for the property. This must be unique per node, and follow the Homie
+    ///   [ID format](https://homieiot.github.io/specification/#topic-ids).
+    /// * `name`: The human-readable name of the property.
+    /// * `settable`: Whether the property can be set by the Homie controller. This should be true
+    ///   for properties like the brightness or power state of a light, and false for things like
+    ///   the temperature reading of a sensor.
+    /// * `unit`: The unit for the property, if any. This may be one of the
+    ///   [recommended units](https://homieiot.github.io/specification/#property-attributes), or
+    ///   any other custom unit.
+    pub fn boolean(id: &str, name: &str, settable: bool, unit: Option<&str>) -> Property {
+        Property::make(id, name, Datatype::Boolean, settable, unit, None::<String>)
+    }
+
+    /// Create a new string property with the given attributes.
+    ///
+    /// # Arguments
+    /// * `id`: The subtopic ID for the property. This must be unique per node, and follow the Homie
+    ///   [ID format](https://homieiot.github.io/specification/#topic-ids).
+    /// * `name`: The human-readable name of the property.
+    /// * `settable`: Whether the property can be set by the Homie controller. This should be true
+    ///   for properties like the brightness or power state of a light, and false for things like
+    ///   the temperature reading of a sensor.
+    /// * `unit`: The unit for the property, if any. This may be one of the
+    ///   [recommended units](https://homieiot.github.io/specification/#property-attributes), or
+    ///   any other custom unit.
+    pub fn string(id: &str, name: &str, settable: bool, unit: Option<&str>) -> Property {
+        Property::make(id, name, Datatype::String, settable, unit, None::<String>)
+    }
+
+    /// Create a new enum property with the given attributes.
+    ///
+    /// # Arguments
+    /// * `id`: The subtopic ID for the property. This must be unique per node, and follow the Homie
+    ///   [ID format](https://homieiot.github.io/specification/#topic-ids).
+    /// * `name`: The human-readable name of the property.
+    /// * `settable`: Whether the property can be set by the Homie controller. This should be true
+    ///   for properties like the brightness or power state of a light, and false for things like
+    ///   the temperature reading of a sensor.
+    /// * `unit`: The unit for the property, if any. This may be one of the
+    ///   [recommended units](https://homieiot.github.io/specification/#property-attributes), or
+    ///   any other custom unit.
+    /// * `format`: The possible values for the enum.
+    pub fn enumeration(
+        id: &str,
+        name: &str,
+        settable: bool,
+        unit: Option<&str>,
+        format: &[&str],
```

Comment:
Is this the right type to use? I tried to do something like `impl IntoIterator<&'a str>` as recommended by https://rust-lang.github.io/api-guidelines/flexibility.html#c-generic but I couldn't get it to work with the example passing in a slice.

Reply:
Sounds okay according to the seed style guide, I think. I suppose if someone has a Vec<String> then they will need to do some conversion, so **maybe** intoiter would be better, but I'm not always convinced by super-generic interfaces.

> Use references where possible - e.g. rather use &str instead of String; or &[T] instead of Vec.
-- https://seed-rs.org/0.8.0/rust

Let me take a quick look at the example but I think this seems fine.

Reply:
Yeah, this feels right. You should always be passing in a slice of static strings to this method, so this will always be the most convenient type.

Ship it.

Reply:
Done.